doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+
+Code and Documentation Style Guide - The Missing Bits
+=====================================================
+
+This is a collection of coding and documentation guidelines for SciPy that
+are not explicitly stated in the existing guidelines and standards, including
+
+* `PEP-8 <https://www.python.org/dev/peps/pep-0008>`_ Style Guide for Python Code
+* `PEP-257 <https://www.python.org/dev/peps/pep-0257>`_ Docstring Conventions
+* `NumPy docstring standard <https://numpydoc.readthedocs.io/en/latest/format.html>`_
+
+Some of these are trivial, and might not seem worth discussing, but in many
+cases, the issue has come up in a pull request review in either the SciPy
+or NumPy repositories.  If a style issue is important enough that a reviewer
+will require a change before merging, then it is important enough to be
+documented--at least for cases where the issue can be resolved with a simple
+rule.
+
+
+Coding Style and Guidelines
+---------------------------
```

Comment:
My common gripe is space after comma in lists and such. We do not enforce it in CI and IMO we should not --- this is a recommendation not a strict rule --- hence these extra guidelines feel like a perfect place for this. Suggested phrasing may be along the lines:

```
When writing lists and enumerations, please make sure to use spaces after commas.

Yes: `a = np.array([1, 2, 3], dtype=float)`
No: `a = np.array([1,2,3],dtype=float)`
```

Reply:
This is already part of PEP-8, so it isn't really a "missing bit".

> We do not enforce it in CI and IMO we should not...

As a well-documented item in PEP-8, I think we *should* enforce it in CI, but with so much legacy code that does not conform to the standard, it is difficult to do.

Reply:
We could use black once and flag the commit for git blame (as per black's [doc](https://github.com/psf/black#migrating-your-code-style-without-ruining-git-blame)). Then the whole repo would be cleaned up.

Reply:
Re black and/or CI : IIRC that switch in pyflakes et al is way too strict and makes some algorithmic code less readable. There certainly were multiple discussions about that.

Reply:
If this is a concern, there is always the option to use things like `# fmt: on/off` for instance.

Reply:
These comments, with a sole reason being fending off over-eager linters look like line noise to a human. 
My two cents anyway

Reply:
Agreed, and maybe there are better ways like a general config. But the price being almost zero comments about style in review makes me want to dive in 😅

Reply:
I'm "resolving" this thread; to `black` or not to `black` is not something we should try to resolve in this PR.